cowamm/profitability/competitor_kpis/pancakeswap/pancakeswap_largest_pools_4232597.sql

```diff
@@ -0,0 +1,64 @@
+-- Computes the TVL for every Pancakeswap pool
+-- Then returns the top {{number_of_pools}} pools by TVL
+-- Input: blockchain, number_of_pools to return
+
+with
+-- finds the pools which have been active since 2024-10-01
+data as (
+    select
+        p.contract_address as pool_address,
+        p.call_block_time as time,
+        p.output__reserve0 as balance0,
+        p.output__reserve1 as balance1,
+        rank() over (partition by p.contract_address order by p.call_block_time desc) as latest
+    from pancakeswap_v2_{{blockchain}}.PancakePair_call_getReserves as p
+    where date_trunc('day', p.call_block_time) >= date_trunc('day', cast('2024-10-01' as date))
+),
+
+--Gets the token0 and token1 addresses for each pool
+t0 as (
+    select
+        contract_address,
+        max(output_0) as token0
+    from pancakeswap_v2_{{blockchain}}.PancakePair_call_token0
+    group by contract_address
+),
+
+t1 as (
+    select
+        contract_address,
+        max(output_0) as token1
+    from pancakeswap_v2_{{blockchain}}.PancakePair_call_token1
+    group by contract_address
+),
+
+--computes the tvl for each pool
+-- for each pool we could get multiple balance values if the function was called multiple times in a same block
+-- we arbitrarily choose the maximum value for each pool
```

Comment:
This seems indeed quite arbitrary. Shouldn't we build a ledger with tvl per transaction and then simply pick the most recent one? I would like for our logic to be consistent across pools as this will require a future maintainer to understand less subtleties.

Reply:
Same explanation as previous comment. 
We could worry about recreating an exact value for each AMM but the code and the execution would be much longer.

Reply:
How big is the performance penalty compared to building a proper tvl ledger? I worry that e.g. for pools with on large LP when they remove all of their liquidity the pool stays indexed with their old large tvl value.

If we use a method that we know is inaccurate. let's very clearly document this in the code. Otherwise it just looks as if we think relying on these `_call_` tables is ok (which it generally isn't)